src/Core/Grid/Definition/Factory/DeleteActionTrait.php

```diff
@@ -36,20 +36,34 @@
  */
 trait DeleteActionTrait
 {
-    protected function buildDeleteAction(string $deleteRouteName, string $deleteRouteParamName, string $deleteRouteParamField): RowActionInterface
-    {
+    protected function buildDeleteAction(
+        string $deleteRouteName,
+        string $deleteRouteParamName,
+        string $deleteRouteParamField,
+        array $extraRouteParams = [],
+        array $translations = []
+    ): RowActionInterface {
         return (new SubmitRowAction('delete'))
             ->setName($this->trans('Delete', [], 'Admin.Actions'))
             ->setIcon('delete')
             ->setOptions([
                 'route' => $deleteRouteName,
                 'route_param_name' => $deleteRouteParamName,
                 'route_param_field' => $deleteRouteParamField,
-                'confirm_message' => $this->trans('Are you sure you want to delete the selected item?', [], 'Admin.Notifications.Warning'),
+                'extra_route_params' => $extraRouteParams,
+                'confirm_message' => array_key_exists('confirm_message', $translations) ?
+                    $translations['confirm_message'] :
+                    $this->trans('Are you sure you want to delete the selected item?', [], 'Admin.Notifications.Warning'),
```

Comment:
Should we keep it singular? Issue #17847 talks about bulk action.

Reply:
@sowbiba, I just have this question, otherwise it's fine for me. :-)

Reply:
If we can have a generic message for both Bulk and Single deletion, it's fine for me ;)

Reply:
Something like **Delete selected items?**

Reply:
```suggestion
                    $this->trans('Are you sure you want to delete the selected item(s)?', [], 'Admin.Global'),
```

Reply:
Let's apply this behavior to every grid, this way it will be more consistent for the users. :-)